main.py

```diff
@@ -0,0 +1,420 @@
+                continue
+
+            # Read the TMDB ID MKV tag
+            try:
+                tags = MKVTags.fromstring(extract_result.stdout)
+            except Exception:
+                LOG.exception('failed to parse tag xml from %s: %s', mkv_path.as_posix(), extract_result.stdout)
+                continue
+            tmdb_id = tags.tmdb_id()
+            if not tmdb_id:
+                continue
+
+            LOG.info('%s = %s', mkv_path.as_posix(), tmdb_id)
+
+            # Extract the tmdb.json for this attachment
+            identify_result = subprocess.run(['mkvmerge', '--identify', mkv_path.as_posix()], capture_output=True)
+            if identify_result.returncode == 0:
+                # Find the attachment by name
+                for line in identify_result.stdout.decode('utf-8').split('\n'):
+                    # Sort out which attachment is tmdb.json
+                    m = re.match(r"Attachment ID (?P<attachment_id>[0-9]+): type '(?P<mime_type>[^']+)',(.*,?) file name '(?P<filename>[^']+)'.*", line)
+                    if m and m.group('filename') == 'tmdb.json':
+                        attachment_id = int(m.group('attachment_id'))
+                    else:
+                        continue
+
+                    # Extract the tmdb.json attachment
+                    tmdb_extract_result = subprocess.run(['mkvextract', '--quiet', mkv_path.as_posix(), 'attachments', f'{attachment_id}:/dev/stdout'], capture_output=True)
+                    if tmdb_extract_result.returncode == 0:
+                        # Parse the tmdb.json
+                        tmdb_details = json.load(io.BytesIO(tmdb_extract_result.stdout))
+
+                        # Stop looking at attachments since the tmdb.json was found and extracted
+                        break
+                else:
+                    LOG.warning('no tmdb.json attachment found for %s, ignoring mkv', mkv_path.as_posix())
+                    continue
+
+            movies[tmdb_id] = {'path': mkv_path, 'tmdb': tmdb_id, 'tmdb_details': tmdb_details}
+
+        self.movies = movies
+
+
+class HandlerFactory(object):
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    def request_handler(self, *args, **kwargs):
+        full_kwargs = dict(self.kwargs)
+        full_kwargs.update(kwargs)
+        return Handler(*args, **full_kwargs)
+
+
+class Handler(http.server.BaseHTTPRequestHandler):
+    protocol_version = 'HTTP/1.1'
+    server_version = 'duckflix-media-server/0.1'
+
+    def __init__(self, *args, library=None, allowed_clients=[], **kwargs):
+        self.allowed_clients = allowed_clients
+        self.library = library
+        return http.server.BaseHTTPRequestHandler.__init__(self, *args, **kwargs)
+
+    def do_GET(self):
+        self.method = 'GET'
+        return self.serve()
+
+    def do_HEAD(self):
+        self.method = 'HEAD'
+        return self.serve()
+
+    # TODO: Overwrite send_error to also send CORS headers
```

Comment:
For something more than an MVP, I think flask or something would be nice.

Reply:
We can talk about this, but that's actually quite difficult given the level of control I need over headers and other finer details of the HTTP response. I tried building this using something like flask and ran into lots of problems with the framework messing up my responses. For the video streaming we will likely continue to need raw control over the responses.

Reply:
https://github.com/duckflix/media-server/issues/4